src/nci_util.c

```diff
@@ -912,6 +1040,173 @@ nci_discovery_ntf_copy(
     }
 }
 
+/*==========================================================================*
+ * Interface
+ *==========================================================================*/
+
+NciModeParam*
+nci_mode_param_copy(
+    const NciModeParam* param,
+    NCI_MODE mode)
+{
+    if (G_LIKELY(param)) {
+        NciModeParam* copy = NULL;
+        gsize size = nci_mode_param_size(param, mode);
+
+        switch (mode) {
+        case NCI_MODE_ACTIVE_POLL_A:        /* fallthrough */
+        case NCI_MODE_PASSIVE_POLL_A:       /* fallthrough */
+        case NCI_MODE_ACTIVE_POLL_F:        /* fallthrough */
+        case NCI_MODE_PASSIVE_POLL_F:       /* fallthrough */
+        case NCI_MODE_PASSIVE_POLL_B:       /* fallthrough */
+        case NCI_MODE_ACTIVE_LISTEN_F:      /* fallthrough */
+        case NCI_MODE_PASSIVE_LISTEN_F:     /* fallthrough */
+            if (size) {
+                copy = g_malloc0(size);
+                (void) nci_mode_param_copy_impl(copy, param, mode);
+            }
+            return copy;
+        case NCI_MODE_PASSIVE_POLL_15693:
+        case NCI_MODE_PASSIVE_LISTEN_15693:
+            break;
+        case NCI_MODE_PASSIVE_LISTEN_A:     /* fallthrough */
+        case NCI_MODE_PASSIVE_LISTEN_B:     /* fallthrough */
```

Comment:
ETOOMANYFALLTHROUGHS. I'm ok with those only if they are necessary to silence a compilation warnings.